packages/blitz/src/middleware.ts

```diff
@@ -17,7 +26,9 @@ export async function handleRequestWithMiddleware<
   } = {},
 ) {
   if (!(res as unknown as MiddlewareResponse).blitzCtx) {
-    ;(res as unknown as MiddlewareResponse).blitzCtx = {} as Ctx
+    ;(res as unknown as MiddlewareResponse).blitzCtx = {
+      revalidatePage: revalidateFactory(res as unknown as NextApiResponse),
```

Comment:
Can we avoid this assertion here? Since we already have a type assertion to `MiddlewareResponse` this feels kinda hacky

Reply:
Technically it would be better to abstract this whole thing to our `MiddlewareResponse` and move this logic into `@blitzjs/next` since this is NEXT specific (so converting all `MiddlewareResponse` to be `NextApiResponse` compliant seams bad since it's gonna make it harder for us to adopt i.e. nuxt/sveltkit...).

I'd keep this for now and handle this with better type overrides when we switch to true GENERIC TOOLKIT instead of NextToolkit